azure-pipelines.yml

```diff
@@ -1,157 +1,155 @@
-    $ANDROID_HOME/platform-tools/adb wait-for-device
-    while [[ $? -ne 0 ]]; do sleep 1; $ANDROID_HOME/platform-tools/adb shell pm list packages; done;
-    $ANDROID_HOME/platform-tools/adb devices
-    echo "Emulator started"
-
-
-    echo "Installing Appium"
-    node --version
-
-    npm list --depth 0
-    npm install -g appium@v1.17.0-rc.1
-    # --chromedriver-skip-install
-    ln -fs /usr/local/lib/node_modules/appium/build/lib/main.js /usr/local/bin/appium
-    chmod +x /usr/local/bin/appium
-    export PATH=$PATH:/usr/local/bin/appium
-
-    echo "Installing and Running Appium doctor"
-    npm install -g appium-doctor
-    ln -fs /usr/local/lib/node_modules/appium-doctor/appium-doctor.js /usr/local/bin/appium-doctor
-    chmod +x /usr/local/bin/appium-doctor
-    export PATH=$PATH:/usr/local/bin/appium-doctor
-
-    npm list --depth 0
-
-    appium --version
-    appium-doctor
-
-    #sudo xcode-select -s /Applications/Xcode_$(XCODE_VERSION).app/Contents/Developer
-    #xcrun simctl list
-    
-    #appium --allow-insecure chromedriver_autodownload &
-    #echo "Appium server started"
-
-- task: Maven@3
-  displayName: 'Run tests using appium local service'
-  inputs:
-    mavenPomFile: 'pom.xml'
-    goals: 'test -DisRemote=false'
-    publishJUnitResults: true
-    testResultsFiles: '**/surefire-reports/TEST-*.xml'
-    javaHomeOption: 'JDKVersion'
-    mavenVersionOption: 'Default'
-    mavenAuthenticateFeed: false
-    effectivePomSkip: false
-    sonarQubeRunAnalysis: false
-
-- script: |
-    appium --allow-insecure chromedriver_autodownload &
-    echo "Appium server started"
-
-- task: Maven@3
-  displayName: 'Run tests using manually started (remote) server'
-  inputs:
-    mavenPomFile: 'pom.xml'
-    goals: 'test -DisRemote=true'
-    publishJUnitResults: true
-    testResultsFiles: '**/surefire-reports/TEST-*.xml'
-    javaHomeOption: 'JDKVersion'
-    mavenVersionOption: 'Default'
-    mavenAuthenticateFeed: false
-    effectivePomSkip: false
-    sonarQubeRunAnalysis: false
-
-- task: CopyFiles@2
-  displayName: 'Copy failure screenshots and test logs'
-  inputs:
-    SourceFolder: '$(Build.SourcesDirectory)/target'
-    Contents: |
-        surefire-reports/failure_screenshots/*.png
-        log/*.log
-    TargetFolder: '$(Build.ArtifactStagingDirectory)'
-  condition: succeededOrFailed()
-
-- task: PublishBuildArtifacts@1
-  displayName: 'Publish copied artifacts'
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'drop'
-    publishLocation: 'Container'
-  condition: succeededOrFailed()
\ No newline at end of file
+steps:
+  - task: SonarCloudPrepare@1
+    displayName: 'Prepare SonarCloud analysis'
+    inputs:
+      SonarCloud: 'SonarCloud'
+      organization: 'aqualityautomation'
+      scannerMode: 'CLI'
+      configMode: 'file'
+      extraProperties: |
+          sonar.coverage.exclusions=**/**
```

Comment:
do we really need to change the pipeline in this PR? How is this related to ScreenFactory?

Reply:
I just formatted it